### PR TITLE
Fixed breaking protected blocks when standing outside of zone/plot. F…

### PR DESCRIPTION
**Pull request by grundyboy34**
Fixes three bugs.
1) Blocks can be broken within a zone when the player is standing outside of the zone.
2) Zones can be overwritten due to id collisions when defining a zone/plot.
3) Explosions causing crashes.

### CODE REVIEW

**Comment by spacebuilder2020** [comment]
Looks good so far.  However, these are not trivial changes so I will need some time to test and fully review these.

File: src/main/java/com/forgeessentials/api/permissions/WorldZone.java [hunk 1]
```diff
@@ -51,6 +51,29 @@ public void afterLoad()
...
+                return zone;
+            }
+        }
+        return null;
+    }
+
+    public int nextAreaZoneId()
+    {
+        int desiredID = getServerZone().getMaxZoneID() + 1;
+        while (desiredID < 5 || getServerZone().getAreaZone(desiredID) != null)
```
Comment (spacebuilder2020) [inline src/main/java/com/forgeessentials/api/permissions/WorldZone.java]: What is the reasoning for this logic?  Specifically the desiredID < 5.

Comment (grundyboy34) [reply src/main/java/com/forgeessentials/api/permissions/WorldZone.java]: Sorry, I should've commented that part. In practice, getAreaZone() doesn't return values when dealing with the server zone( id 1 I believe), or the world zone (id 2 I believe). So, this forces it to proceed until at least id of 5 to account for those, and to leave space just incase other such hardcoded zones are added later.

Comment (spacebuilder2020) [reply src/main/java/com/forgeessentials/api/permissions/WorldZone.java]: Ok.  Going forward, I would recommend not using hardcoded values for constants.  For example, `desiredID < 5` should be replaced with `desiredID < RESERVED_ID_COUNT`, where reserved_id_count is defined as `private static final int RESERVED_ID_COUNT = 5;`.  For this specific line, the risk is low and the logic is sound so I don't think you need to replace this value with a constant, but as a best practice, it is good to use constants in place of hardcoded values.

**Comment by grundyboy34** [comment]
Also noticed that explosions were causing crashes since the Explosion class's fields and the ReflectionHelper's getPrivateValue method have changed, this second commit fixes those.

**Comment by spacebuilder2020** [comment]
I am going to hold on merging this for now as I have not had time to review or test any of these changes.

**Review by spacebuilder2020 — COMMENTED** [review]
1) Approved
2) Rejected due to being too big
3) Rejected as use of Access Transformers is recommended over using ReflectionHelper

Since 2/3 of the changes were rejected, I am forced to reject this pull.  I would recommend following the new [rules](https://github.com/ForgeEssentials/ForgeEssentials#development-setup) for PRs and re-implement changes 1 and 2 as separate PRs.  Change 3 was already implemented my most recent commit.

**Pull request closed without merging**